src/WebJobs.Script/Workers/Rpc/RpcInitializationService.cs

```diff
@@ -127,79 +98,22 @@ internal async Task InitializeRpcServerAsync()
             }
             catch (Exception grpcInitEx)
             {
-                var hostInitEx = new HostInitializationException($"Failed to start Rpc Server. Check if your app is hitting connection limits.", grpcInitEx);
+                throw new HostInitializationException($"Failed to start Rpc Server. Check if your app is hitting connection limits.", grpcInitEx);
             }
         }
 
         internal Task InitializeChannelsAsync()
         {
-            // TODO: Remove special casing when resolving https://github.com/Azure/azure-functions-host/issues/4534
-            if (ShouldStartAsPlaceholderPool())
+            if (_environment.IsPlaceholderModeEnabled())
             {
-                return _webHostRpcWorkerChannelManager.InitializeChannelAsync(_workerRuntime);
-            }
-            else if (ShouldStartStandbyPlaceholderChannels())
-            {
-                return InitializePlaceholderChannelsAsync();
-            }
-
-            return InitializeWebHostRuntimeChannelsAsync();
-        }
-
-        private Task InitializePlaceholderChannelsAsync()
-        {
-            if (_environment.IsLinuxAzureManagedHosting())
-            {
-                return InitializePlaceholderChannelsAsync(OSPlatform.Linux);
-            }
-
-            return InitializePlaceholderChannelsAsync(OSPlatform.Windows);
-        }
-
-        private Task InitializePlaceholderChannelsAsync(OSPlatform os)
-        {
-            return Task.WhenAll(_hostingOSToRuntimesAllowList[os].Select(runtime =>
-                _webHostRpcWorkerChannelManager.InitializeChannelAsync(runtime)));
-        }
-
-        private Task InitializeWebHostRuntimeChannelsAsync()
-        {
-            if (_webHostLeveldRuntimesAllowList.Contains(_workerRuntime, StringComparer.OrdinalIgnoreCase))
-            {
-                return _webHostRpcWorkerChannelManager.InitializeChannelAsync(_workerRuntime);
-            }
-
-            return Task.CompletedTask;
-        }
-
-        internal bool ShouldStartStandbyPlaceholderChannels()
-        {
-            if (string.IsNullOrEmpty(_workerRuntime) && _environment.IsPlaceholderModeEnabled())
-            {
-                if (_environment.IsLinuxAzureManagedHosting())
+                if (_placeholderLanguageWorkersList != null && _placeholderLanguageWorkersList.Count() != 0)
```

Comment:
I think we can remove the null check for _placeholderLanguageWorkersList. also lets log the list of runtimes to start in placeholder mode even if list is empty (for alerting).

Reply:
done!